File: app/src/main/java/za/co/kernelpanic/cloudy/data/Temperature.java
package za.co.kernelpanic.cloudy.data;

import com.google.gson.annotations.SerializedName;


public class Temperature {

    @SerializedName("dt")
    private long date;
    @SerializedName("day")
    private double dayForecast;
    @SerializedName("min")
    private double minTemp;
    @SerializedName("max")
    private double maxTemp;


    public long getDate() {
        return date;
    }

    public double getDayForecast() {
        return dayForecast;
    }

    public double getMinTemp() {
        return minTemp;
    }

    public double getMaxTemp() {
        return maxTemp;
    }


    //The constructor below is used by room
    public Temperature(long date, double dayForecast, double minTemp, double maxTemp) {

        this.date = date;
        this.dayForecast = dayForecast;
        this.minTemp = minTemp;
        this.maxTemp = maxTemp;
    }
}

File: app/src/main/java/za/co/kernelpanic/cloudy/data/Temperature.kt
package za.co.kernelpanic.cloudy.data

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Temperature(@Json(name = "dt") val date: Long? = 0,
                       @Json(name = "day") val dayForecast: Double? = 0.0,
                       @Json(name = "min") val minTemp: Double? = 0.0,
                       @Json(name = "max") val maxTemp: Double? = 0.0)